packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,442 @@
+        defines.MULTIPLYDEBUG = options.multiplyDebug ?? defines.MULTIPLYDEBUG;
+        super(material, "MeshDebug", 200, defines);
+
+        options.mode = defines.MODE;
+        options.multiplyDebug = defines.MULTIPLYDEBUG;
+        options.shadedDiffuseColor = options.shadedDiffuseColor ?? new Vector3(1, 1, 1);
+        options.shadedSpecularColor = options.shadedSpecularColor ?? new Vector3(0.8, 0.8, 0.8);
+        options.shadedSpecularPower = options.shadedSpecularPower ?? 10;
+        options.wireframeThickness = options.wireframeThickness ?? 0.7;
+        options.wireframeColor1 = options.wireframeColor1 ?? new Vector3(0, 0, 0);
+        options.wireframeColor2 = options.wireframeColor2 ?? new Vector3(0.9, 0.9, 0.9);
+        options.vertexRadius = options.vertexRadius ?? 1.2;
+        options.uvScale = options.uvScale ?? 20;
+        options.uvColor1 = options.uvColor1 ?? new Vector3(1, 1, 1);
+        options.uvColor2 = options.uvColor2 ?? new Vector3(0.5, 0.5, 0.5);
+        options.materialColor = options.materialColor ?? this._getRandomColor();
+        this._options = options;
+        this._mode = this._options.mode!;
+
+        this._enable(this._isEnabled);
+    }
+
+    /**
+     * Get the class name
+     * @returns Class name
+     */
+    public getClassName() {
+        return "MeshDebugPluginMaterial";
+    }
+
+    /**
+     * Prepare the defines
+     * @param defines Mesh debug defines
+     * @param _scene Scene
+     * @param _mesh Mesh associated with material
+     */
+    public prepareDefines(defines: MeshDebugDefines, scene: Scene, mesh: AbstractMesh) {
+        if ((this._mode == MeshDebugMode.VERTICES || this._mode == MeshDebugMode.TRIANGLES) && !mesh.isVerticesDataPresent("initialPass")) {
+            Logger.Warn("For best results with VERTICES or TRIANGLES mode, please use MeshDebugPluginMaterial.tripleTriangles() on mesh.", 1);
+        }
+        defines.MODE = this._mode;
+        defines.MULTIPLYDEBUG = this._options.multiplyDebug!;
+    }
+
+    /**
+     * Get the shader attributes
+     * @param attributes Array of attributes
+     */
+    public getAttributes(attr: string[]) {
+        attr.push("initialPass");
+    }
+
+    /**
+     * Get the shader uniforms
+     * @returns Uniforms
+     */
+    public getUniforms() {
+        return {
+            ubo: [
+                { name: "m_shadedDiffuseColor", size: 3, type: "vec3" },
+                { name: "m_shadedSpecularColor", size: 3, type: "vec3" },
+                { name: "m_shadedSpecularPower", size: 1, type: "float" },
+                { name: "m_wireframeThickness", size: 1, type: "float" },
+                { name: "m_wireframeColor1", size: 3, type: "vec3" },
+                { name: "m_wireframeColor2", size: 3, type: "vec3" },
+                { name: "m_vertexRadius", size: 1, type: "float" },
+                { name: "m_uvScale", size: 1, type: "float" },
+                { name: "m_uvColor1", size: 3, type: "vec3" },

```

Comment:
To save some room in the uniform buffer (we already use quite a lot variables in the PBR material!), you could use a single `vec4` for the 4 `m_shadedSpecularPower`, `m_wireframeThickness`, `m_vertexRadius` and `m_uvScale` floats.